src/safeposix/syscalls/fs_calls.rs

```diff
@@ -63,15 +57,15 @@ impl Cage {
                     atime: time, ctime: time, mtime: time,
                 });
 
-                let newinodenum = mutmetadata.nextinode;
-                mutmetadata.nextinode += 1;
-                if let Inode::Dir(ind) = mutmetadata.inodetable.get_mut(&pardirinode).unwrap() {
+                let newinodenum = FS_METADATA.nextinode.load(interface::RustAtomicOrdering::Relaxed);
+                FS_METADATA.nextinode.store(newinodenum + 1 as usize, interface::RustAtomicOrdering::Relaxed);
```

Comment:
Yeah. While the rest of the PR makes most of these operations so much easier to read, this is now harder. I shouldn't really have to think to determine that this is a simple increment.

Reply:
This is where atomic_add should be used instead